hevc/sps.go

```diff
@@ -181,3 +183,73 @@ func (s *SPS) ImageSize() (width, height uint32) {
 	height = encHeight - (s.ConformanceWindow.TopOffset+s.ConformanceWindow.BottomOffset)*subHeightC
 	return width, height
 }
+
+// CodecString returns string based on SPS fields.
+// ISO/IEC 14496-15:2014 Annex E.
+func (s *SPS) CodecString() string {
+
+	fields := []string{"hvc1"}
+	generalProfileSpace := ""
+	switch s.ProfileTierLevel.GeneralProfileSpace {
+	case 1:
+		generalProfileSpace = "A"
+	case 2:
+		generalProfileSpace = "B"
+	case 3:
+		generalProfileSpace = "C"
+	}
+
+	fields = append(fields, fmt.Sprintf("%s%d", generalProfileSpace, s.ProfileTierLevel.GeneralProfileIDC))
+
+	profileCompatibilityFlags := reverseBitsAndHexEncode(s.ProfileTierLevel.GeneralProfileCompatibilityFlags)
+	fields = append(fields, profileCompatibilityFlags)
+
+	profileGeneralTier := "L"
+	if s.ProfileTierLevel.GeneralTierFlag {
+		profileGeneralTier = "H"
+	}
+
+	fields = append(fields, fmt.Sprintf("%s%d", profileGeneralTier, s.ProfileTierLevel.GeneralLevelIDC))
+
+	constraints := make([]byte, 8)
+	binary.BigEndian.PutUint64(constraints, s.ProfileTierLevel.GeneralConstraintIndicatorFlags)
+	end := 8
+	for {
+		if constraints[end-1] != 0 {
+			break
+		}
+		end--
+	}
+	fields = append(fields, trimLeadingZero(fmt.Sprintf("%x", constraints[0:end])))
+
+	return strings.Join(fields, ".")
+
+}
+
+// reverseBitsAndHexEncode encodes the 32 bits input, but in reverse bit order
+// ISO/IEC 23008‐2
+func reverseBitsAndHexEncode(x uint32) string {
+
+	x = ((x & 0x55555555) << 1) | ((x & 0xAAAAAAAA) >> 1)
```

Comment:
This way of reversing the bits of each individual byte in the uint32 value is a bit hard to understand compared to the simplistic approach in PR #72. It is probably more efficient, though.